cardano-node/src/Cardano/CLI/Ops.hs

```diff
@@ -151,6 +170,57 @@ instance Show CliError where
   show (ReadSigningKeyFailure fp expt)
     = "Exception encountered while trying to read the signing key file at: " <> fp
        <> "Exception: " <> T.unpack expt
-  show (InvariantViolation err)
-    = "Internal invariant violated: " <> err
+
 instance Exception CliError
+
+data RealPBFTError =
+    IncorrectProtocolSpecified !Protocol
+  | FromProtocolError !ProtocolInstantiationError
+  | GenesisBenchmarkRunnerError !TxGenError
+  | InvariantViolation !Text
+  | TransactionTypeNotHandledYet !Text
+  deriving Show
+
+data TxGenError =
+    CurrentlyCannotSendTxToRelayNode !FilePath
+  -- ^ Relay nodes cannot currently be transaction recipients.
+  | InsufficientFundsForRecipientTx
+  -- ^ Error occurred while creating the target node address.
+  | NeedMinimumThreeSigningKeyFiles ![FilePath]
+  -- ^ Need at least 3 signing key files.
+  | SecretKeyDeserialiseError !Text
+  | SecretKeyReadError !Text
+  deriving Show
+
+-- | Perform an action that expects ProtocolInfo for Byron/PBFT,
+--   with attendant configuration.
+withRealPBFT
+  :: Text
+  -> Maybe NodeId
+  -> Maybe Int
+  -> GenesisFile
+  -> RequiresNetworkMagic
+  -> Maybe Double
+  -> Maybe DelegationCertFile
+  -> Maybe SigningKeyFile
+  -> Update
+  -> Protocol
+  -> (RunNode ByronBlock
+        => Consensus.Protocol ByronBlock
+        -> ExceptT RealPBFTError IO a)
+  -> ExceptT RealPBFTError IO a
+withRealPBFT gHash nId mNumNodes genFile nMagic sigThresh delCertFp sKeyFp update ptcl action = do
+  SomeProtocol p <- firstExceptT FromProtocolError $ fromProtocol
```

Comment:
This should be indented so the subsequent lines don't depend on the `SomeProtocol p <- firstExceptT FromProtocolError`'s length.